src/types/response/guards.ts

```diff
@@ -23,3 +28,50 @@ export const isProcessedSingleAnswerResponse = (
     isProcessedFieldResponse(response)
   )
 }
+
+export const isProcessedCheckboxResponse = (
+  response: any,
+): response is ProcessedCheckboxResponse => {
+  return (
+    'answerArray' in response &&
+    Array.isArray(response.answerArray) &&
+    // Check that all elements in answerArray are string (including empty string)
+    response.answerArray.every((elem: any) => typeof elem === 'string') &&
+    isProcessedFieldResponse(response)
+  )
+}
+
+export const isTableRow = (row: any): row is ITableRow => {
+  return (
+    // Check that the row contains a single array of only string (including empty string)
+    Array.isArray(row) && row.every((elem: any) => typeof elem === 'string')
+  )
+}
+
+export const isProcessedTableResponse = (
+  response: any,
+): response is ProcessedTableResponse => {
+  if (
+    'answerArray' in response &&
+    Array.isArray(response.answerArray) &&
+    // Check that all elements in answerArray are table rows
+    response.answerArray.every((arr: any) => isTableRow(arr))
+  ) {
+    // Check that all arrays in answerArray have the same length
+    const subArrLength: number = response.answerArray[0].length
```

Comment:
is it possible that `response.answerArray[0]` is `undefined`?

Reply:
thanks for pointing out, i've added an additional check that response.answerArray[0] must be a tableRow